classes/core/AltapaySettings.php

```diff
@@ -1,392 +1,409 @@
+		} else {
+			$message = '';
+		}
+
+		$_SESSION['altapay_capture_failed'] = $message . $newMessage;
+	}
+
+	/**
+	 * @param string $newMessage
+	 * @return void
+	 */
+	public function saveCaptureWarning( $newMessage ) {
+		if ( isset( $_SESSION['altapay_capture_warning'] ) ) {
+			$message = $_SESSION['altapay_capture_warning'] . '<br/>';
+		} else {
+			$message = '';
+		}
+
+		$_SESSION['altapay_capture_warning'] = $message . $newMessage;
+	}
+
+	/**
+	 * Displays login error message
+	 *
+	 * @return void
+	 */
+	public function loginError() {
+		$this->showUserMessage( 'altapay_login_error', 'error', 'Could not login to the Merchant API: ' );
+	}
+
+	/**
+	 * @param string $field
+	 * @param string $type
+	 * @param string $message
+	 * @return void
+	 */
+	public function showUserMessage( $field, $type, $message = '' ) {
+		$this->startSession();
+
+		if ( ! isset( $_SESSION[ $field ] ) ) {
+			return;
+		}
+
+		echo "<div class='$type notice'> <p>$message $_SESSION[$field]</p> </div>";
+
+		unset( $_SESSION[ $field ] );
+	}
+
+	/**
+	 * Displays failed capture message
+	 *
+	 * @return void
+	 */
+	public function captureFailed() {
+		$this->showUserMessage( 'altapay_capture_failed', 'error' );
+	}
+
+	/**
+	 * Displays warning message against capture request
+	 *
+	 * @return void
+	 */
+	public function captureWarning() {
+		$this->showUserMessage( 'altapay_capture_warning', 'update-nag' );
+	}
+
+	/**
+	 * @param array $links
+	 * @return array
+	 */
+	public function addActionLinks( $links ) {
+		$newLink = array(
+			'<a href="' . admin_url( 'admin.php?page=altapay-settings' ) . '">Settings</a>',
+		);
+		return array_merge( $links, $newLink );
+	}
+
+
+	/**
+	 * Loads language file with language specifics
+	 *
+	 * @return void
+	 */
+	public function altapayLocalizationInit() {
+		 load_plugin_textdomain( 'altapay', false, dirname( plugin_basename( __FILE__ ) ) . '/languages/' );
+	}
+
+	/**
+	 * Add AltaPay settings option in plugins menu
+	 *
+	 * @return void
+	 */
+	public function altapaySettingsMenu() {
+		add_submenu_page(
+			'woocommerce',
+			'AltaPay Settings',
+			'AltaPay Settings',
+			'manage_options',
+			'altapay-settings',
+			array( $this, 'altapaySettings' )
+		);
+	}
+
+	/**
+	 * Register AltaPay specific settings group including url and api login credentials
+	 *
+	 * @return void
+	 */
+	public function altapayRegisterSettings() {
+		 register_setting( 'altapay-settings-group', 'altapay_gateway_url' );
+		register_setting( 'altapay-settings-group', 'altapay_username' );
+		register_setting( 'altapay-settings-group', 'altapay_password' );
+		register_setting( 'altapay-settings-group', 'altapay_payment_page' );
+		register_setting( 'altapay-settings-group', 'altapay_terminals_enabled', 'json_encode' );
+	}
+
+	/**
+	 * AltaPay settings page with actions and controls
+	 *
+	 * @return void
+	 * @throws AltapayMerchantAPIException
+	 */
+	public function altapaySettings() {
+		 $terminals        = false;
```

Comment:
Incorrect indentation
```suggestion
		$terminals        = false;
```